### ISSUE TITLE
Celery Workers stall and do nothing

### ISSUE THREAD

**Issue by akashkumar58** [celery/celery#4025]
## Details
I am using `Celery 4.0.2` with `Redis 3.2.1` in a django project. I create a group of tasks and execute them asynchronously, this **works fine when the number of tasks in that group is 1 less than the number of CPUs(workers prefork). As soon as number of tasks in that group becomes greater or equal to number of celery workers, nothing happens. All workers stall**. Redis queue length remains 0 unless I kill celery workers. After killing celery workers it queues all the tasks with giving output as follows:

> Restoring 4 unacknowledged message(s)

When I restart my celery workers, they consume all the messages from redis. I have tried using `Rabbitmq` also but the same thing happens. As there is no error messages I am not able to understand the problem. Any help is much appreciated.

**Comment by eburmicz** [comment]
I have this issue too.

**Comment by eburmicz** [comment]
How did you fix it?

**Comment by auvipy** [comment]
make sure you are using the latest release and can reproduce this with celery master

**Comment by akashkumar58** [comment]
@eburmicz I had worked on this a long time back. If I remember correctly, there was a flaw in my design of tasks itself. The way I was creating the group of tasks, there was a dependency in-between tasks and that is what was leading to a deadlock in case of number tasks was greater than the number of CPU cores. When I had fixed the dependency issue, the tasks started to work as expected. I forgot that I had created a github issue for the same.
